src/error.cs

```diff
@@ -203,4 +214,32 @@ static bool ShowInternal (Exception e)
 		}
 		return error;
 	}
+
+	public static WarningLevel GetWarningLevel (int code)
```

Comment:
Maybe we could (later, not in this PR) use partial classes and move the common/identical code into a single file, keeping the differences in each product.